sklearn/linear_model/_ridge.py

```diff
@@ -556,6 +558,15 @@ def _ridge_regression(
             # we implement sample_weight via a simple rescaling.
             X, y = _rescale_data(X, y, sample_weight)
 
+    if not isinstance(alpha, (np.ndarray, type(None), tuple)):
+        alpha = check_scalar(
+            alpha,
+            "alpha",
+            target_type=numbers.Real,
+            min_val=0.0,
+            include_boundaries="left",
+        )
```

Comment:
Was this change was introduced for `RidgeCV`?

If so, I would reintroduce it in another PR and move the `check_scalar` call under the one for `self.max_iter` in `fit`.

If not, I would add this comment to explain its intend:
```suggestion
    # Some callers of this method might pass alpha as single
    # element array which already has been validated.
    if not isinstance(alpha, (np.ndarray, type(None), tuple)):
        alpha = check_scalar(
            alpha,
            "alpha",
            target_type=numbers.Real,
            min_val=0.0,
            include_boundaries="left",
        )
```